farmfs/snapshot.py

```diff
@@ -68,17 +76,21 @@ def __iter__(self):
     udd = self.udd
     exclude = self.exclude
     def tree_snap_iterator():
-      for entry, type_ in root.entries(exclude):
-        tree_path = entry.relative_to(root)
+      last_path = None
+      for path, type_ in root.entries(exclude):
+        tree_str = path.relative_to(root)
+        if last_path:
+          assert last_path < path, "Order error: %s < %s" % (last_path, Path)
+        last_path = path
         if type_ == "link":
-          ud_path = entry.readlink().relative_to(udd)
+          ud_str = path.readlink().relative_to(udd)
         elif type_ == "dir":
-          ud_path = None
+          ud_str = None
         elif type_ == "file":
           continue
         else:
           raise ValueError("Encounted unexpected type %s for path %s" % (type_, entry))
-        yield SnapshotItem(tree_path, type_, ud_path, reverser=self.reverser)
+        yield SnapshotItem(tree_str, type_, ud_str, reverser=self.reverser)
```

Comment:
It would be nice if SnapshotItems could get out of the reverser game and just get the checksums they want.